release how-to-guide managing cherry-picks (#62)
aviator-co/gitbook-docs: releases-beta/how-to-guides/managing-cherry-picks.md

> # Managing Cherry-Picks

Reviewer: Please also add a link in the summary page to reference this doc

> Cherry-picking is an essential tool in version control systems, allowing you to apply specific commits from one branch to another without merging the entire branch. In Aviator’s release management system, cherry-picks are used to selectively integrate changes into release candidates. This guide will walk you through the process of managing cherry-picks in Aviator, ensuring smooth handling of your release process.

Reviewer: ```suggestion
Cherry-picking enables you to apply specific commits from one branch to another without merging the entire branch. In Aviator’s release management system, cherry-picks are used to selectively integrate changes into release candidates. This guide will walk you through the process of managing cherry-picks in Aviator, ensuring smooth handling of your release process.
```

## Prerequisites

> Before you begin cherry-picking, ensure that your repository is properly connected to Aviator and that a release project has been created.

Reviewer: May be link to: https://docs.aviator.co/releases-beta/getting-started

1. Set up your repository and connect it to Aviator.

> ![](../../.gitbook/assets/release-github-integration.png)

Reviewer: This is the wrong image, this one connects a GitHub user handle to Aviator. We probably don't need image for this, this action is done during the onboarding.

2. [Create a release project](./creating-a-release-project.md) if it does not already exist.

## Cherry-Picking a Pull Request (PR)

> ### Step 1: Navigate to the Aviator Release Dashboard

Reviewer: ```suggestion
### Step 1: Cut a new release
```

1. Open Aviator’s Release Dashboard.

> 2. Cut a release from the listed PRs on the dashboard.

Reviewer: Mention that this is the release on which you will apply a cherry-pick. So one can skip this step if they already have an existing release to base cherry-pick on.

![](../../.gitbook/assets/release-conflict-base-cut.png)

### Step 2: Cherry-Pick the PR

1. Find the specific PR you wish to cherry-pick into your release.

2. Click on the PR to open the detailed view.

> 3. Once inside the PR details, locate the Cherry-Pick button.

Reviewer: ```suggestion
3. Click the "Cherry-pick" button in the PR details view.
```

![](../../.gitbook/assets/release-cherry-pick-button.png)

> 2. Click the button to initiate the cherry-pick process. Aviator will apply the changes from the PR into the selected release candidate in the dropdown.

Reviewer: This is the most important part that you completely skipped over.

Aviator enables users to apply same PR to multiple release candidates. You can select the oldest release version, and Aviator will apply cherry-pick to all versions newer than that.

- Users can also deselect the versions they don't want to apply cherry-picks on
- Aviator will automatically detect which versions already include this PR and will skip the cherry-pick to those versions

Reviewer: Please also include the image with multiple versions getting applied.

> 3. If the cherry-pick is successful, a second release candidate of the release will be created.

Reviewer: Mention that it may take a few seconds to apply the cherry-pick. Talk about all the states: pending, success and failure with screenshots for each.

![](../../.gitbook/assets/release-cherry-pick-details.png)

## Cherry-Pick Failures

In some cases, a cherry-pick may fail due to conflicts between the changes in the PR and the current state of the release branch. Aviator will flag such conflicts and provide tools to help resolve them.

![](../../.gitbook/assets/release-conflict-cherry-pick.png)

> To learn how to resolve cherry-pick failures, refer to the detailed guide on [Resolving Cherry-Pick Failures](./resolving-a-cherry-pick-failure.md).

Reviewer: Move this text above the image.
